pkg/reconciler/nscert/nscert.go

```diff
@@ -141,6 +143,8 @@ func (c *reconciler) reconcile(ctx context.Context, ns *corev1.Namespace) error
 	} else if !equality.Semantic.DeepEqual(existingCert.Spec, desiredCert.Spec) {
 		copy := existingCert.DeepCopy()
 		copy.Spec = desiredCert.Spec
+		copy.ObjectMeta.Labels = desiredCert.ObjectMeta.Labels
```

Comment:
Instead of overwriting all of the labels, I would prefer only overwriting the `WildcardCertDomainLabelKey` label.
If users add their own labels for their own purpose, we probably still want to keep them. WDYT?